### PR TITLE
feat: add intermediates directory. Copy file when get plugins and build

### PR DESCRIPTION
**Pull request by befovy**
copy files from plugin's go/dlib folder to intermediates folder
when 'hover plugins get'.

copy files in intermediates to binary output folder before build.

implement and finish #296

### CODE REVIEW

**Comment by befovy** [comment]
this pr implement https://github.com/go-flutter-desktop/go-flutter/issues/296

File: cmd/build.go [hunk 1]
```diff
@@ -532,13 +533,17 @@ func buildNormal(targetOS string, vmArguments []string) {
 
 func buildEnv(targetOS string, engineCachePath string) []string {
 	var cgoLdflags string
+
+	outputDirPath := build.OutputDirectoryPath(targetOS)
+
 	switch targetOS {
 	case "darwin":
 		cgoLdflags = fmt.Sprintf("-F%s -Wl,-rpath,@executable_path", engineCachePath)
+		cgoLdflags = fmt.Sprintf("%s -F%s -L%s", cgoLdflags, outputDirPath, outputDirPath)
```
Comment (jld3103) [inline cmd/build.go]: Why are these flags removed: `-Wl,-rpath,@executable_path`? Are they not necessary anymore? Did you test?

File: internal/build/build.go [hunk 2]
```diff
@@ -11,11 +11,10 @@ import (
 // Much like android and ios are already used.
 const BuildPath = "go"
 
-// OutputDirectoryPath returns the path where the go-flutter binary and flutter
-// binaries blobs will be stored for a particular platform.
+// buildDirectoryPath returns the path in `BuildPath`/build.
 // If needed, the directory is create at the returned path.
-func OutputDirectoryPath(targetOS string) string {
-	outputDirectoryPath, err := filepath.Abs(filepath.Join(BuildPath, "build", "outputs", targetOS))
+func buildDirectoryPath(targetOS, path string) string {
```
Comment (jld3103) [inline internal/build/build.go]: Missing type for `targetOS`

Comment (pchampio) [reply internal/build/build.go]: In golang `targetOS, path string` == `targetOS string, path string`

File: cmd/build.go (hunk 1 above)
Comment (pchampio) [reply cmd/build.go]: It's not removed, the changes adds -F and -L flags to the existing `cgoLdflags` ;)

**Comment by pchampio** [comment]
Oups forgot, `hover plugins tidy --purge` should delete the go/build/intermediates

**Comment by befovy** [comment]
Please take a look.
Add the missing `dlib`  operator for `hover init-plugin` and `hover plugins tidy --purge`

**Comment by pchampio** [comment]
Awesome! Thank for your contribution.

**Pull request merged**